addon/utils/serialize-query-params.js

```diff
@@ -0,0 +1,65 @@
+const RBRACKET = /\[\]$/;
+
+function isPlainObject(obj) {
+  return Object.prototype.toString.call(obj) === '[object Object]';
+}
+
+/*
+ * Ported from "@ember-data/adapter/-private/utils/serialize-query-params"
+ * Helper function that turns the data/body of a request into a query param string.
+ * This is directly copied from jQuery.param.
+ */
+export default function serializeQueryParams(queryParamsObject) {
+  var s = [];
+
+  function buildParams(prefix, obj) {
+    var i, len, key;
+
+    if (prefix) {
+      if (Array.isArray(obj)) {
+        for (i = 0, len = obj.length; i < len; i++) {
+          if (RBRACKET.test(prefix)) {
+            add(s, prefix, obj[i]);
+          } else {
+            buildParams(prefix + '[' + (typeof obj[i] === 'object' ? i : '') + ']', obj[i]);
+          }
+        }
+      } else if (isPlainObject(obj)) {
+        for (key in obj) {
+          buildParams(prefix + '[' + key + ']', obj[key]);
+        }
+      } else {
+        add(s, prefix, obj);
+      }
+    } else if (Array.isArray(obj)) {
+      for (i = 0, len = obj.length; i < len; i++) {
+        add(s, obj[i].name, obj[i].value);
+      }
+    } else {
+      for (key in obj) {
+        buildParams(key, obj[key]);
+      }
+    }
+    return s;
+  }
+
+  return buildParams('', queryParamsObject)
+    .join('&')
+    .replace(/%20/g, '+');
+}
```

Comment:
Thanks for porting this over.  It's a lot clearer now and we don't have the dep.

Looking at the specific implementation though, I think we should use a much simpler version that only handles primitive values (so no array handling or object handling).

Users can still use those values, but they would have to put them in the URL themselves, or encode them as strings before passing them to `urlParams`.

The reason I don't want this implementation in particular is that there isn't a standard encoding scheme for arrays and nested structures to a URL, and the right encoding scheme will depend on the API.  So I don't think we should use any particular encoding, beyond stuff that will work with more or less all APIs (i.e. simple `${key}=${value}`).

So this means that things in params that wind up in the content get `toJSON` and things that wind up in the URL as query params get `toString` which seems like a reasonable mental model for users IMHO.